### Eclipse Workspace Patch 1.0
#P SLandsbek_SimpleOrderExport
Index: src/app/code/community/SLandsbek/SimpleOrderExport/Model/Export/Csv.php
===================================================================
--- src/app/code/community/SLandsbek/SimpleOrderExport/Model/Export/Csv.php	(revision 10)
+++ src/app/code/community/SLandsbek/SimpleOrderExport/Model/Export/Csv.php	(working copy)
@@ -128,6 +128,7 @@
             'Shipping State Name',
             'Shipping Country',
             'Shipping Country Name',
+            'Shipping Phone Number',
     		'Billing Name',
             'Billing Company',
             'Billing Street',
@@ -137,6 +138,7 @@
             'Billing State Name',
             'Billing Country',
             'Billing Country Name',
+            'Billing Phone Number',
             'Order Item Increment',
     		'Item Name',
             'Item Status',
@@ -195,6 +197,7 @@
             $shippingAddress ? $shippingAddress->getRegion() : '',
             $shippingAddress ? $shippingAddress->getCountry() : '',
             $shippingAddress ? $shippingAddress->getCountryModel()->getName() : '',
+            $shippingAddress ? $shippingAddress->getData("telephone") : '',
             $billingAddress->getName(),
             $billingAddress->getData("company"),
             $billingAddress->getData("street"),
@@ -203,7 +206,8 @@
             $billingAddress->getRegionCode(),
             $billingAddress->getRegion(),
             $billingAddress->getCountry(),
-            $billingAddress->getCountryModel()->getName()
+            $billingAddress->getCountryModel()->getName(),
+            $billingAddress->getData("telephone")
         );
     }
 
